### PR TITLE
[11.0] [FIX] Payment Reference Visibility

### PR DESCRIPTION
**Pull request by Tatider**
Always show "Payment Reference Number" for Customers
and Vendors invoices.

### CODE REVIEW

**Comment by aisopuro** [comment]
@mtnarjan the lint is failing:
```
l10n_fi_payment_reference/models/account_invoice.py:119:21: E126 continuation line over-indented for hanging indent
```

**Comment by aisopuro** [comment]
@pedrobaeza the tests are failing because of failing to connect to the database:
```
Testing ['l10n_fi_banks', 'l10n_fi_business_code', 'l10n_fi_business_code_validate', 'l10n_fi_payment_reference']:

createdb: could not connect to database template1: could not connect to server: No such file or directory

	Is the server running locally and accepting

	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
```
Is there someone we can contact to help us with this?

**Comment by pedrobaeza** [comment]
That's because a change in Travis infrastructure. Fixed in https://github.com/OCA/l10n-finland/commit/9d3d8f2d5983e391b89c71ff201c2ef7b3501aa0.

Please rebase over new 11.0 branch

**Comment by aisopuro** [comment]
The coverage check seems to be failing because 10n_fi_payment_reference` doesn't have any tests that make sure that validating invoices actually generates the payment reference as expected. A new test should be added for doing that.

**Comment by Tatider** [comment]
No, % of coverage is the same and test for it is passed (greed highlight).
But there is no new test for new changes (red highlight), what impossible in this case because:
1. Change is in xml
2. Another changes in code style, to pass style guide check pass.
So no test can be added for changes
Screenshot from 2019-10-01 11-26-06

**Comment by aisopuro** [comment]
The problem is basically that when the code was originally added, this particular coverage check seems to not have been in place. That means that the implementation of `invoice_validate` was accepted even though it was not covered by tests. However, now that is no longer OK, and so a test covering `invoice_validate` needs to be added, since it was changed.

**Comment by aisopuro** [comment]
Rebased commits to be clearer about changes.

**Review by aisopuro — APPROVED** [review]
Looks good.

File: l10n_fi_payment_reference/models/account_invoice.py [hunk 1]
```diff
@@ -115,7 +115,10 @@ def _compute_payment_reference(self):
     def invoice_validate(self):
         res = super(AccountInvoiceFinnish, self).invoice_validate()
         # noinspection PyProtectedMember
-        self.filtered(lambda i: i.payment_reference_type != 'none' and
-                      i.type == 'out_invoice') \
-            ._compute_payment_reference()
+        self.filtered(
+            lambda i:
+                i.payment_reference_type != 'none'
```
Comment (mlaitinen) [inline l10n_fi_payment_reference/models/account_invoice.py:120]: Please add parentheses. This kind of combination of ANDs and ORs without them is confusing.

**Review by mlaitinen — CHANGES_REQUESTED** [review]
Just a small change to make the code more understandable.

File: l10n_fi_payment_reference/models/account_invoice.py (hunk 1 above)
Comment (Tatider) [reply l10n_fi_payment_reference/models/account_invoice.py:120]: Fixed.

**Review by mlaitinen — APPROVED** [review]

**Pull request merged**